runtime/jvmti/jvmtiHook.c

```diff
@@ -1030,8 +1036,7 @@ jvmtiHookClassPrepare(J9HookInterface** hook, UDATA eventNum, void* eventData, v
 
 	/* Call the event callback */
 
-	if (callback != NULL) {
-		J9VMThread * currentThread = data->currentThread;
+	if (callback != NULL && shouldPostEvent(currentThread, NULL)) {
```

Comment:
```suggestion
	if ((NULL != callback) && shouldPostEvent(currentThread, NULL)) {
```

Reply:
This pattern appears numerous places: please fix them all.